base/promotion.jl

```diff
@@ -222,9 +222,8 @@ minmax(x::Real, y::Real) = minmax(promote(x, y)...)
 # for the multiplication of two types,
 #   promote_op{R<:MyType,S<:MyType}(::typeof(*), ::Type{R}, ::Type{S}) = MyType{multype(R,S)}
 promote_op(::Any)    = (@_pure_meta; Bottom)
-promote_op(::Any, T) = (@_pure_meta; T)
-promote_op{T}(::Type{T}, ::Any) = (@_pure_meta; T)
```

Comment:
`promote_op{T}(::Type{T}, ::Any) = (@_pure_meta; T)` was added in #16179 without things like `Float32.([1,2,3])` will return `Any` and that will fallback to the new slow broadcast algorithm. I think it it's convenient to keep it. Of course, I'll leave it for you to consider it.